atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleQueryFactory.java

```diff
@@ -352,17 +353,17 @@ protected FullQuery getRowsColumnRangeSubQuery(
             long ts,
             BatchColumnRangeSelection columnRangeSelection) {
         String query = " /* GET_ROWS_COLUMN_RANGE (" + tableName + ") */ "
-                + "SELECT * FROM ( SELECT m.row_name, m.col_name, max(m.ts) as ts"
+                + "SELECT s.row_name, s.col_name, s.ts" + getValueSubselect("s", true)
+                + " FROM ( SELECT m.row_name, m.col_name, max(m.ts) as ts"
+                +         getValueSubselectForGroupBy("m")
```

Comment:
So I experimented with applying the same idea to other KVS calls, including getLatestCells() and it actually gets slower. We get rid of the extra self-join which is good, but unfortunately Oracle performs a `SORT GROUP BY` instead of `HASH GROUP BY` when it sees `DENSE_RANK LAST ORDER BY ts`, which kills the perf. With the new query, the plan looks something like this:
```
SORT GROUP BY
`-NESTED LOOPS
    |- FULL SCAN on t
    `- INDEX RANGE SCAN on m
```
While with the existing query, it's like this:
```
NESTED LOOPS
|-HASH GROUP BY
|   `-NESTED LOOPS
|      |-FULL SCAN on t
|      `-INDEX RANGE SCAN on m
`-INDEX UNIQUE SCAN on m
```
It seems there's no optimizer hint to force `HASH GROUP BY`.